### PR TITLE
feat: health check for openshift DeploymentConfig

### PR DESCRIPTION
**Pull request by ishitasequeira**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Signed off by [Ishita Sequeira](@ishitasequeira )

### CODE REVIEW

File: resource_customizations/apps.openshift.io/DeploymentConfig/health.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    elseif numTrue == 1 then
+      health_check.status = "Progressing"
+      return health_check
+    else
+      health_check.status = "Degraded"
+      health_check.message = "Deployment config is degraded"
+      return health_check
+    end
+  end
+end
```
Comment (jannfis) [inline resource_customizations/apps.openshift.io/DeploymentConfig/health.lua:27]: if `obj.status` is `nil`, we'll return nothing. Please initialize `health_check` and return it in `Progressing` state for this case.

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks @ishitasequeira. I have one comment, please see below.

File: resource_customizations/apps.openshift.io/DeploymentConfig/health.lua (hunk 1 above)
Comment (ishitasequeira) [reply resource_customizations/apps.openshift.io/DeploymentConfig/health.lua]: Thank you @jannfis for the review. I have added the Progressing status.

**Review by jannfis — APPROVED** [review]

**Comment by jannfis** [comment]
LGTM, thank you @ishitasequeira

**Comment by fredcb** [comment]
I really like thes feature and just disovered it, but I can't find info about how to use it. Is there any reference to this somewhere that I missed?

**Pull request merged**